app/src/main/java/org/mifos/mobile/ui/registration/RegistrationVerificationScreen.kt

```diff
@@ -0,0 +1,169 @@
+    ) {
+
+    val context = LocalContext.current
+
+    var requestID by rememberSaveable(stateSaver = TextFieldValue.Saver) {
+        mutableStateOf(TextFieldValue(""))
+    }
+    var authenticationToken by rememberSaveable(stateSaver = TextFieldValue.Saver) {
+        mutableStateOf(TextFieldValue(""))
+    }
+
+    Column(modifier = Modifier.fillMaxSize()) {
+        when (uiState) {
+
+            RegistrationUiState.Initial -> {
+                Column(modifier = Modifier.fillMaxSize()) {
+
+                    MifosOutlinedTextField(
+                        value = requestID,
+                        onValueChange = { requestID = it },
+                        label = R.string.request_id,
+                        supportingText = "",
+                        keyboardType = KeyboardType.Number
+                    )
+
+                    MifosOutlinedTextField(
+                        value = authenticationToken,
+                        onValueChange = { authenticationToken = it },
+                        label = R.string.authentication_token,
+                        supportingText = "",
+                        keyboardType = KeyboardType.Number
+                    )
+
+                    Button(
+                        onClick = {
+                            verifyUser(authenticationToken.toString(), requestID.toString())
+                        },
+                        Modifier
+                            .fillMaxWidth()
+                            .padding(start = 16.dp, end = 16.dp, top = 4.dp),
+                        contentPadding = PaddingValues(12.dp),
+                        colors = ButtonDefaults.buttonColors(
+                            containerColor = if (isSystemInDarkTheme()) Color(
+                                0xFF9bb1e3
+                            ) else Color(0xFF325ca8)
+                        )
+                    ) {
+                        Text(text = stringResource(id = R.string.verify))
+                    }
+                }
+            }
+
+             is RegistrationUiState.Error -> {
+                Toast.makeText(context, uiState.exception, Toast.LENGTH_SHORT).show()
+                 MifosErrorComponent(
+                     isNetworkConnected = Network.isConnected(context),
+                     isEmptyData = false,
+                     isRetryEnabled = true,
+                     onRetry = { retryConnection(authenticationToken.toString(), requestID.toString()) }
+                 )
+            }
+
+            RegistrationUiState.Loading -> {
+                MifosProgressIndicator(
+                    modifier = Modifier
+                        .fillMaxSize()
+                        .padding(10.dp)
+                        .background(MaterialTheme.colorScheme.background.copy(alpha = 0.7f))
+                )
+            }
+
+             RegistrationUiState.Success -> {
+                context.startActivity(Intent(context, LoginActivity::class.java))
```

Comment:
dont start activity from here, do this in fragment, the callback should call this LoginActivity from fragment